sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -343,4 +363,11 @@ object ResolveDefaultColumns {
       v1Catalog.isPersistentFunction(ident.asFunctionIdentifier)
     }
   }
+
+  trait CustomInsertSchema {
```

Comment:
Move this under package `org.apache.spark.sql.connector.write` as `SupportsCustomSchemaWrite`? 
We will need the implement a new class for testing the new trait(interface) as well

Reply:
@gengliangwang it needs to be visible from the `ResolveDefaultColumns` rule which is under `package org.apache.spark.sql.catalyst.analysis`. That `package org.apache.spark.sql.connector.write` is not visible in this way. Is there a different package we should use instead?

I will add a test table that inherits this trait, and exercise it in the unit tests and comment here again when ready.

Reply:
It should be accessible. 
https://github.com/apache/spark/tree/master/sql/catalyst/src/main/java/org/apache/spark/sql/connector/write

Reply:
Thanks for the pointer, I was able to implement this suggestion.